src/App.tsx

```diff
@@ -1,39 +1,82 @@
-import React from 'react';
+import { Component } from 'react';
 import './App.scss';
+import { Clock } from './Clock';
 
 function getRandomName(): string {
   const value = Date.now().toString().slice(-4);
 
   return `Clock-${value}`;
 }
 
-export const App: React.FC = () => {
-  const today = new Date();
-  let clockName = 'Clock-0';
+type State = {
+  hasClock: boolean;
+  clockName: string;
+};
+
+export class App extends Component<{}, State> {
+  state: Readonly<State> = {
+    hasClock: true,
+    clockName: 'Clock-0',
+  };
+
+  timerId = 0;
+
+  componentDidMount() {
+    this.timerId = window.setInterval(() => {
+      this.setState({ clockName: getRandomName() });
+    }, 3300);
+
+    document.addEventListener('contextmenu', (event) => {
+      event.preventDefault();
+    });
+
+    document.addEventListener('mouseup', (e) => {
+      if (e.button === 0) {
+        this.handleLeftBtnMouse();
+      }
 
-  // This code starts a timer
-  const timerId = window.setInterval(() => {
-    clockName = getRandomName();
-  }, 3300);
+      if (e.button === 2) {
+        this.handleRightBtnMouse();
+      }
+    });
+  }
 
-  // this code stops the timer
-  window.clearInterval(timerId);
+  componentWillUnmount(): void {
+    // this code stops the timer
+    window.clearInterval(this.timerId);
 
-  return (
-    <div className="App">
-      <h1>React clock</h1>
+    document.removeEventListener('contextmenu', (event) => {
+      event.preventDefault();
+    });
 
-      <div className="Clock">
-        <strong className="Clock__name">
-          {clockName}
-        </strong>
+    document.removeEventListener('mouseup', (e) => {
+      if (e.button === 0) {
+        this.handleLeftBtnMouse();
+      }
 
-        {' time is '}
+      if (e.button === 2) {
+        this.handleRightBtnMouse();
+      }
```

Comment:
Here you are repeating code from addEventListener, consider moving reusable logic to separate functions